### PR TITLE
storage: implement proper sematic for RawCAS (#9895)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #9895 to release-5.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tikv repo:
git pr https://github.com/tikv/tikv/pull/9907
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tikv.git pr/9907:release-5.0-75a0e1efc46f
```

---

### What problem does this PR solve?

When comparison failed, the previous RawCAS implementation can't distinguish the previous value between empty value and none value.

[kvproto changes](https://github.com/pingcap/kvproto/pull/742)

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

- Renamed `compare_and_set` to `compare_and_swap`
- In the `RawCASRespond`, return `succeed` instead `not_equal`
- Add `previous_not_exist` to `RawCASRespond`

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

File: src/server/service/kv.rs [hunk 1]
```diff
@@ -322,6 +323,19 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
         ctx.spawn(task);
     }
 
+    fn coprocessor_v2(
```
Comment (Little-Wallace) [inline src/server/service/kv.rs:326]: It seems to be unfinished method.

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Comment by andylokandy** [comment]
This is supposed to merge to v5.0.1

**Pull request merged**